osf/migrations/0161_add_registration_files_count.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-01-18 14:43
+from __future__ import unicode_literals
+import logging
+import progressbar
+
+from django.db import migrations, models
+from django.db.models import Count
+from django_bulk_update.helper import bulk_update
+from osf.models import Registration
+
+logger = logging.getLogger(__file__)
+
+def add_registration_files_count(state, *args, **kwargs):
+    registrations = Registration.objects.filter(is_deleted=False).filter(
+        files__type='osf.osfstoragefile',
+        files__deleted_on__isnull=True
+    ).annotate(
+        annotated_file_count=Count('files')
```

Comment:
See https://github.com/CenterForOpenScience/osf.io/pull/8915/files#diff-3e3a3f525cf036eaea66f072633d775f for comparison:

# Modifications
1) Casting a wider net - not filtering on ArchiveJobs - found a small percentage of public registrations whose ArchiveJobs are still initiating or they don't have an ArchiveJob at all. (Users can turn them public, is one reason).    Populating `files_count` on more registrations, even if most of those registrations aren't shown seems more thorough.
2) Speed down from 20 minutes to 6 minutes by annotating the queryset with the files count instead of looping through and fetching files count one by one
3) Still using bulk_update to batch so we don't have one gigantic thing going in